### DISCUSSION TITLE
Has anyone gotten any lightweight graphic browser to work with iSH?

### DISCUSSION THREAD

**Discussion by SugarRayLua** [ish-app/ish#1486] [Q&A]
Has anyone had success in getting any browser to work with iSH? I've not had full success with any browser I've tried. I know that iSH currently has limited graphics capabilities, and first tried running a browser with iSH as a VNC server and using a VNC client on my iPad Pro. However, iSH crashed (disconnected VNC) whenever I tried to open any of the graphical browser which I thought might be due to iSH's current limited graphical capabilities. Thus, I switched to trying to open graphical browsers in iSH using iSH as a ssh server and using the Mocha X11 iPad app as an ssh viewer with X11 forwarding. The nice thing about the Mocha X11 app is it takes over all the 2D graphic rending/resources from iSH so it eliminates difficulty with graphic rendering as a reason why apps like graphical web browser might not work (actually renders 2D graphics reasonably quickly). Using the Mocha X11 app as the ssh viewer, I can get the lightweight apps Dillo and Links 2 (using the -g graphical option) to open and graphically display Google's search page. However, when I then use Google to search for a webpage and try to connect to that webpage, iSH crashes giving me an "illegal instruction" error. Thus, I think there may still be some limitation in the iSH emulator preventing the graphical web-browsers from working not related to rendering graphics. I'm not able to try Midori since it requires dbus which I'm aware is not currently functioning in iSH. Has anyone else been successful? Thanks.

**Comment by saagarjha** [comment]
Since this isn't really an issue, I'm going to turn it into a discussion.

**Comment by guanzhangrtk** [comment]
Sorry for hijacking the thread but I'm curious which version of iSH you are running and how you setup the VNC server. I've followed the instructions on the Wiki but could never get a functioning server going, it appears as if nothing is listening on the port and I don't really know how to confirm with existing tools, thanks!

**Reply by SugarRayLua** [reply]
Hi, guanzhangrtk, happy to try and help.

I followed the iSH wiki page to for my setup and it’s been working for me for several iterations of iSH. I’m currently running:

iSH 1.2.2
(iPad Pro, 11in, 3rd Gen, iOS 14.6)

The iPad viewing app that I use simultaneously in split screen when VNC connecting is:
VNC Remoter on the App Store (I think others have used cheaper ones that also worked).
I set up the VNC section of the VNC Remoter app as follows:
Name: my iPad’s name
VNC Hostname: local host
VNC Port: 5900

I have to though make sure I start the VNC server script before I try on log on with the VNC Remoter client or else like has happened to you, it won’t log on. The VNC server script is just the last script in the top section of the Wiki instructions. I named the script “VNC” when I saved it into my home directory, so I know I just need to go to my home directory and type, “./VNC” and then the VNC script opens an X window and the VNC server. I then attempt to log on from my VNC remoter client app.

The only other peculiarity is that if I sometime get improperly disconnected or close the app and go to logon again, I need to repeat those same steps (start the VNC script then logon with the VNC Remoter client) to restart the VNC session.

I hope that helps. Let me know if there is anything else I can answer about my setup.

**Reply by guanzhangrtk** [reply]
Thanks for the response, do you mind installing "VNC Viewer - Remote Desktop" by RealVNC and see if it works (it's free). I'd appreciate it just to rule out any strange issues, thanks!

**Reply by SugarRayLua** [reply]
Sure, guanzhangrtk, I just downloaded it and tried it using the steps I listed above, and it works fine in terms of connecting and viewing a VNC session. In fact, it even displays high-resolution 2D images which my other paid VNC viewer couldn’t, and I thought was a problem with iSH (and even discussed the VNC viewer difficulty in the Reddit forum—after your suggestion, I added an addendum that VNC is a workable 2D rendering solution with the right viewer). The only thing is that the Real VNC viewer is a little difficult to navigate with iSH as opposed to my other viewers— it doesn’t respond to the all of the iPad Pro’s keyboard commands (need to use on screen terminal keys).

Please let me know if there is anything else I can help test to help you get yours working. Have a good night.

**Reply by guanzhangrtk** [reply]
Thanks for the confirmation. I finally got it working, the issue seems to be with the final script. On my iPad, I had to manually run `startx &`, wait a little bit and then run the next command. I guess for some reason that command executed before X was up and running, and therefore was not able to start the VNC server. I guess manually adding some delay in the script probably would work too.

**Reply by SugarRayLua** [reply]
Strange that you had to add that pause but great that you got it working!

**Comment by dustyCheese72** [comment]
You can run text based browsers like eww for emacs. I haven't had any trouble with that but yeah I don't know how to do graphics

**Comment by canandlamp** [comment]
running a basic python http server on iSH, and using a web browser in split screen as the front end pointed to localhost, has enabled graphical interface for me

**Reply by Ilovepandass** [reply]
How did you do that? I’m new to this type of stuff btw
